### PR TITLE
DAH-2879 feat: invalid login banner

### PR DESCRIPTION
**Pull request by cade-exygy**
## Description

Added new string value that includes a link to /forgot-password page
Specifically handles the "bad_credentials" response

## Jira ticket

[2879](https://sfgovdt.jira.com/browse/DAH-2879)

### Version Control

- [x] branch name begins with `angular` if it contains updates to Angular code
- [x] branch name contains the Jira ticket number
- [x] PR name follows `type: TICKET-NUMBER Description` format, e.g. `feat: DAH-123 New Feature`

### Code quality

- [x] [the set of changes is small](https://google.github.io/eng-practices/review/developer/small-cls.html#what-is-small)
- [x] all automated code checks pass (linting, tests, coverage, etc.)
- [x] code irrelevant to the ticket is not modified e.g. changing indentation due to automated formatting
- [x] if the code changes the UI, it matches the UI design exactly
- [ ] if the changes include human translations, follow the [human translations process](https://sfgovdt.jira.com/l/cp/XS1KpvE4)

### Review instructions

- [x] instructions specify which environment(s) it applies to
- [x] instructions work for PA testers
- [x] instructions have already been performed at least once

### Request review

- [x] PR has `needs review` label
- [x] Use `Housing Eng` group to automatically assign reviewers, and/or assign specific engineers
- [x] If time sensitive, notify engineers in Slack

## Review
Go to /sign-in?react=true
Try any invalid email and password
Ensure banner appears and link to forgot-password page works
Ensure you can close the banner successfully
If you enter a valid account, or unconfirmed account, the view reflects that accurately.
Ensure styles for desktop and mobile match figma.

Optional: If you can cause / mock a 503 error: you should see the unknown error banner

### CODE REVIEW

**Comment by tallulahkay** [comment]
@cade-exygy could we make the base branch on the pr the parent branch so the diff is clean for reviewing?

**Comment by cade-exygy** [comment]
> @cade-exygy could we make the base branch on the pr the parent branch so the diff is clean for reviewing?

Ah yes, I forgot that step. I wish github was smart enough to know to point my PR towards the parent branch I initiated the branch from lol

File: app/assets/json/translations/react/en.json [hunk 1]
```diff
@@ -1563,6 +1563,7 @@
   "shortFormNav.review": "Review",
   "shortFormNav.you": "You",
   "signIn.badCredentials": "Invalid login credentials. Please try again.",
+  "signIn.badCredentialsWithResetLink": "Email or password is incorrect. Check for mistakes and try again. <a target='_blank' href='%{url}'>Reset your password</a> if you need to.",
```
Comment (tallulahkay) [inline app/assets/json/translations/react/en.json:1566]: Do we have confirmation from design that we want this to open in a new window/tab?

File: app/javascript/authentication/SignInForm.tsx [hunk 2]
```diff
@@ -139,6 +140,22 @@ const SignInForm = () => {
...
+    if (error.response.data.error === "not_confirmed") {
+      setNewAccountNotConfirmedModal(error.response.data.email)
+    } else if (error.response.data.error === "bad_credentials") {
+      setRequestError(
+        t("signIn.badCredentialsWithResetLink", {
+          url: getForgotPasswordPath(),
+        })
+      )
+    } else {
+      // TODO: handle sign-in error states
```
Comment (tallulahkay) [inline app/javascript/authentication/SignInForm.tsx:153]: are there a tickets we can link these TODO's to?

File: app/javascript/authentication/SignInForm.tsx [hunk 3]
```diff
@@ -139,6 +140,22 @@ const SignInForm = () => {
   )
   const { signIn } = useContext(UserContext)
 
+  const handleRequestError = (error: AxiosError<{ error: string; email: string }>) => {
```
Comment (tallulahkay) [inline app/javascript/authentication/SignInForm.tsx:143]: I like this abstraction to a function!

File: app/assets/json/translations/react/en.json (hunk 1 above)
Comment (cade-exygy) [reply app/assets/json/translations/react/en.json:1566]: I will ask!

File: app/javascript/authentication/SignInForm.tsx (hunk 2 above)
Comment (cade-exygy) [reply app/javascript/authentication/SignInForm.tsx:153]: Yeah, I am waiting to hear back from design for the 503 one

File: app/assets/json/translations/react/en.json (hunk 1 above)
Comment (cade-exygy) [reply app/assets/json/translations/react/en.json:1566]: will change to same tab

File: app/javascript/authentication/SignInForm.tsx (hunk 2 above)
Comment (cade-exygy) [reply app/javascript/authentication/SignInForm.tsx:153]: will add 503 error message

File: app/assets/json/translations/react/en.json [hunk 4]
```diff
@@ -1578,6 +1580,12 @@
   "signIn.signInRequired": "Please sign in to your account to access this page.",
   "signIn.signedOutSuccessfully": "You have successfully logged out of your account.",
   "signIn.userTokenValidationTimeout": "We care about your security. We logged you out due to a connection issue. Please sign in to continue.",
+  "signIn.newAccount.title" : "Check your email to finish creating your account",
+  "signIn.newAccount.p1": "We sent a link to %{email}.",
+  "signIn.newAccount.p2": "Follow the link in your email to finish creating your account. It expires after 24 hours.",
+  "signIn.newAccount.sendEmailAgainButton": "Send email again",
+  "signIn.newAccount.sendEmailAgainButton.confirmation": "Email sent. Check your email.",
+  "signIn.newAccount.sendEmailAgainButton.error": "Something went wrong. We could not send an email. Try sending it again or refreshing the page.",
```
Comment (chadbrokaw) [inline app/assets/json/translations/react/en.json:1588]: These strings are duplicated?

Comment (cade-exygy) [reply app/assets/json/translations/react/en.json:1588]: hmm let me try merging latest main

Comment (cade-exygy) [reply app/assets/json/translations/react/en.json:1588]: weird, I will delete them, and then run the phrase commands again

**Comment by cade-exygy** [comment]
> Hey ! SO the sign in button automatically redirects to home
> 
>  Screen.Recording.2024-12-04.at.2.09.38.PM.mov

hmm, its a weird issue. Might be due to a bad merge, troubleshooting

**Pull request closed without merging**